core/dbt/cli/requires.py

```diff
@@ -37,6 +40,14 @@ def wrapper(*args, **kwargs):
             flags.DEBUG,
         )
 
+        # Now that we have our logger, fire away!
+        fire_event(MainReportVersion(version=str(installed_version), log_version=LOG_VERSION))
+        flags_dict_str = cast_dict_to_dict_of_strings(get_flag_dict())
+        fire_event(MainReportArgs(args=flags_dict_str))
```

Comment:
The values needs to be stringified first, because it's defined as `Dict[str, str]`:
https://github.com/dbt-labs/dbt-core/blob/f1087e57bfbcc39f793d76b98464550f330090d8/core/dbt/events/types.proto#L82-L85

(The good news is, this was caught by some of our functional tests!)

As discussed during BLG on Friday, we should investigate if we can support the `Struct` proto type (`Dict[str, Any]`): https://github.com/dbt-labs/dbt-core/issues/6803#issuecomment-1426138432